storage/innobase/buf/buf0lru.cc

```diff
@@ -986,16 +986,16 @@ buf_LRU_block_free_non_file_page(
 	ut_ad(!block->page.in_LRU_list);
 	ut_ad(!block->page.hash);
 
-	block->page.set_state(buf_page_t::NOT_USED);
-
-	MEM_UNDEFINED(block->page.frame, srv_page_size);
```

Comment:
I do not see any reason to move this code. `buf_page_t::set_state()` should remain a simple assignment of `block->page.zip.fix`.